pkg/network/cache/client.go

```diff
@@ -27,9 +30,19 @@ func NewSDKClientCache(opts ...client.Option) *ClientCache {
 }
 
 // Get function returns existing client or creates a new one.
-func (c *ClientCache) Get(address string) (client.Client, error) {
+func (c *ClientCache) Get(netAddr *network.Address) (client.Client, error) {
+	hostAddr, err := netAddr.HostAddrString()
```

Comment:
Since `hostAddr` is not used as key for cache map, you can move this code down near `opts := append(c.opts, client.WithAddress(hostAddr))`

Reply:
moved